apps/src/aiComponentLibrary/copyButton/CopyButton.tsx

```diff
@@ -0,0 +1,46 @@
+import Button from '@code-dot-org/component-library/button';
+import React, {useState} from 'react';
+
+import style from './copy-button.module.scss';
+
+// Fallback method for browsers that do not support navigator.clipboard
+const copyToClipboard = (text: string) => {
+  const textField = document.createElement('textarea');
+  textField.innerText = text;
+  document.body.appendChild(textField);
+  textField.select();
+  document.execCommand('copy');
+  textField.remove();
+};
+
+const CopyButton: React.FC<{copyText: string}> = ({copyText}) => {
+  const CONFIRM_TIMEOUT_MS = 1500;
```

Comment:
Should this be a prop (with a default value)?

Reply:
Would we want different timeouts on different copy buttons? The more abstractified thing I was imagining was that we could have a general `ConfirmIconButton` or something that could get used here, but didn't want to get too aggressive with the refactoring:

https://github.com/code-dot-org/code-dot-org/blob/2b640fb43233dcd8bfc3f4c82b1fbc067703e5fe/apps/src/aiDifferentiation/AiDiffBotMessageFooter.tsx#L62-L81

In any case, [Mark asked for some updates](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1738356628501919?thread_ts=1737590475.011849&cid=C06FELVTV0Q) around adding a tooltip to these buttons, so I might end up doing a second pass here in a follow-up if that would work for you?